web/src/search_suggestion.ts

```diff
@@ -729,46 +729,78 @@ function get_operator_suggestions(last: NarrowTerm): Suggestion[] {
     });
 }
 
+// One full search suggestion can include multiple search terms, based on what's
+// in the search bar.
+type SuggestionLine = Suggestion[];
+function suggestion_search_string(suggestion_line: SuggestionLine): string {
+    const search_strings = [];
+    for (const suggestion of suggestion_line) {
+        if (suggestion.search_string !== "") {
+            search_strings.push(suggestion.search_string);
+        }
+    }
+    return search_strings.join(" ");
+}
+
 class Attacher {
-    result: Suggestion[] = [];
+    result: SuggestionLine[] = [];
     prev = new Set<string>();
-    base: Suggestion;
+    base: SuggestionLine;
 
-    constructor(base: Suggestion) {
+    constructor(base: SuggestionLine) {
         this.base = base;
     }
 
-    prepend_base(suggestion: Suggestion): void {
-        if (this.base && this.base.description_html.length > 0) {
-            suggestion.search_string = this.base.search_string + " " + suggestion.search_string;
-            suggestion.description_html =
-                this.base.description_html + ", " + suggestion.description_html;
-        }
-    }
-
-    push(suggestion: Suggestion): void {
-        if (!this.prev.has(suggestion.search_string)) {
-            this.prev.add(suggestion.search_string);
-            this.result.push(suggestion);
+    push(suggestion_line: SuggestionLine): void {
+        const search_string = suggestion_search_string(suggestion_line);
+        if (!this.prev.has(search_string)) {
+            this.prev.add(search_string);
+            this.result.push(suggestion_line);
         }
     }
 
     push_many(suggestions: Suggestion[]): void {
         for (const suggestion of suggestions) {
-            this.push(suggestion);
+            this.push([suggestion]);
         }
     }
 
     attach_many(suggestions: Suggestion[]): void {
         for (const suggestion of suggestions) {
-            this.prepend_base(suggestion);
-            this.push(suggestion);
+            const suggestion_line = [...this.base, suggestion];
+            this.push(suggestion_line);
         }
```

Comment:
Does this add the `base` several times?

Reply:
It does the same thing that it did before, where it first prepends the base to the given `suggestion` and then calls `this.push` with that (base prepended) suggestion. Each suggestion only has `base` prepended once during `attach_many`, and `attach_many` is called in a place where `base` wouldn't have already been prepended to the suggestions.